### PR TITLE
Add new Spec

### PR DESCRIPTION
**Pull request by romefeller**
This PR creates a new Spec (v6) to change the current use of <nixpkgs> to a pinned nixpkgs.

### CODE REVIEW

**Comment by madeline-os** [comment]
What does that mean?

**Comment by madeline-os** [comment]
This needs a changelog

File: src/Nix/Thunk.hs [hunk 1]
```diff
@@ -637,14 +638,31 @@ let fetch = { private ? false, fetchSubmodules ? false, owner, repo, rev, sha256
 in fetch json
 |]
 
+gitHubThunkSpecV6 :: ThunkSpec
```
Comment (madeline-os) [inline src/Nix/Thunk.hs:641]: Needs haddocks explaining why this exists

File: src/Nix/Thunk.hs [hunk 2]
```diff
@@ -729,6 +747,27 @@ let fetch = {url, rev, branch ? null, sha256 ? null, fetchSubmodules ? false, pr
 in fetch json
 |]
 
+gitThunkSpecV6 :: ThunkSpec
```
Comment (madeline-os) [inline src/Nix/Thunk.hs:750]: Haddock

**Review by madeline-os — CHANGES_REQUESTED** [review]
Just needs documentation and changelog

**Comment by plt-amy** [comment]
@madeline-os tested by:

```
$ nix-thunk create https://github.com/plt-amy/1lab.git 1lab
$ cat 1lab/thunk.nix
# DO NOT HAND-EDIT THIS FILE
let fetch = { private ? false, fetchSubmodules ? false, owner, repo, rev, sha256, ... }:
  if !fetchSubmodules && !private then builtins.fetchTarball {
    url = "https://github.com/${owner}/${repo}/archive/${rev}.tar.gz"; inherit sha256;
  } else (builtins.fetchTarball {
  url = "https://github.com/NixOS/nixpkgs/archive/3aad50c30c826430b0270fcf8264c8c41b005403.tar.gz";
  sha256 = "0xwqsf08sywd23x0xvw4c4ghq0l28w2ki22h0bdn766i16z9q2gr";
}).fetchFromGitHub {
    inherit owner repo rev sha256 fetchSubmodules private;
  };
  json = builtins.fromJSON (builtins.readFile ./github.json);
in fetch json⏎
$ nix-thunk unpack 1lab
Cloning into './1lab-7dbbebbff4b4db50/.'...
HEAD is now at 6d51d2ea defn: start work on synthetic homotopy theory; n-truncations
✔ Fetched thunk 1lab
```

then poking around the files to make sure everything was in order, which it was.

**Review by madeline-os — APPROVED** [review]

**Pull request merged**